File: app/src/main/java/com/gamebooster/launcher/GameLauncher.java
package com.gamebooster.launcher;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * Oyunları başlatan ve optimize eden sınıf
 */
public class GameLauncher {
    private Context context;
    private FpsBooster fpsBooster;
    private final String TAG = "GameLauncher";

    public GameLauncher(Context context) {
        this.context = context;
        this.fpsBooster = new FpsBooster(context);
    }

    /**
     * Oyunu FPS optimizasyonu ile başlat
     */
    public LaunchResult launchGameWithBoost(String packageName) {
        try {
            // Sistem optimizasyonu yap
            FpsBooster.OptimizationResult optimizationResult = fpsBooster.optimizeForGame(packageName);

            // Oyunu başlat
            Intent intent = context.getPackageManager().getLaunchIntentForPackage(packageName);

            if (intent != null) {
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                context.startActivity(intent);
                Log.d(TAG, "Game launched: " + packageName);
                return new LaunchResult.Success(optimizationResult.getCleanedRam(),
                        optimizationResult.getExpectedFpsBoost());
            } else {
                Log.e(TAG, "Cannot find launch intent for: " + packageName);
                return new LaunchResult.Error("Oyun başlatılamadı");
            }
        } catch (Exception e) {
            Log.e(TAG, "Launch error: " + e.getMessage());
            return new LaunchResult.Error("Hata: " + e.getMessage());
        }
    }

    /**
     * Oyun başlatma sonucu
     */
    public static abstract class LaunchResult {
        public static class Success extends LaunchResult {
            private long cleanedRam;
            private int expectedFpsBoost;

            public Success(long cleanedRam, int expectedFpsBoost) {
                this.cleanedRam = cleanedRam;
                this.expectedFpsBoost = expectedFpsBoost;
            }

            public long getCleanedRam() {
                return cleanedRam;
            }

            public int getExpectedFpsBoost() {
                return expectedFpsBoost;
            }
        }

        public static class Error extends LaunchResult {
            private String message;

            public Error(String message) {
                this.message = message;
            }

            public String getMessage() {
                return message;
            }
        }
    }
}

File: app/src/main/java/com/gamebooster/launcher/GameLauncher.kt
package com.gamebooster.launcher

import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager

/**
 * Oyunları başlat ve yönet
 */
class GameLauncher(private val context: Context) {

    private val packageManager: PackageManager = context.packageManager
    private val fpsBooster = FpsBooster(context)

    /**
     * Oyunu FPS boost ile başlat
     */
    fun launchGameWithBoost(packageName: String): Result {
        return try {
            // Sistem optimizasyonunu başlat
            val stats = fpsBooster.optimizeForGame(packageName)

            // Intent ile oyunu başlat
            val intent = packageManager.getLaunchIntentForPackage(packageName)
                ?: return Result.Error("Oyun başlatılamadı")

            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
            context.startActivity(intent)

            Result.Success(stats)
        } catch (e: Exception) {
            Result.Error(e.message ?: "Bilinmeyen hata")
        }
    }

    /**
     * Oyun çalışırken gerçek zamanlı metrikleri al
     */
    fun getGamePerformance(): GameMetrics {
        return fpsBooster.getGameMetrics()
    }

    /**
     * Optimizasyonu sıfırla
     */
    fun resetOptimization() {
        fpsBooster.resetSystem()
    }
}

sealed class Result {
    data class Success(val stats: GameStats) : Result()
    data class Error(val message: String) : Result()
}
